servers/rendering/renderer_rd/shaders/effects/taa_resolve.glsl

```diff
@@ -273,7 +278,9 @@ vec3 clip_history_3x3(uvec2 group_pos, vec3 color_history, vec2 velocity_closest
 	// Compute min and max (with an adaptive box size, which greatly reduces ghosting)
 	vec3 color_avg = (s1 + s2 + s3 + s4 + s5 + s6 + s7 + s8 + s9) * RPC_9;
 	vec3 color_avg2 = ((s1 * s1) + (s2 * s2) + (s3 * s3) + (s4 * s4) + (s5 * s5) + (s6 * s6) + (s7 * s7) + (s8 * s8) + (s9 * s9)) * RPC_9;
-	float box_size = mix(0.0f, 2.5f, smoothstep(0.02f, 0.0f, length(velocity_closest)));
+	float variance_scale = 1080.0f / params.resolution.y;
+	float variance = clamp(BASE_VARIANCE * variance_scale, BASE_VARIANCE_MIN, BASE_VARIANCE_MAX); // Variance dynamically scales based on resolution
```

Comment:
I'm not sure about this. I suspect the issue posted by LiveTrower has more to do with having a huge dynamic range than it does with having a lower res screen. Have you tested and confirmed that this makes a difference?

If this indeed makes a difference, then all these calculations should be moved outside the shader since this value will be the same for all pixels.

Reply:
> Nvidia suggests using MSAA with TAA to help with this issue and indeed does work.

While this works, it's a very uncommon approach in modern games due to the GPU cost, so we can't exactly rely on it as a baseline. It's also generally impossible when using deferred rendering (or even slower, in the few engines that support deferred + MSAA).